src/main/java/com/hyperwallet/clientsdk/util/MultipartRequest.java

```diff
@@ -0,0 +1,123 @@
+        buildHeaders();
+        URL url = new URL(requestURL);
+        final String pair = username + ":" + password;
+        final String base64 = DatatypeConverter.printBase64Binary(pair.getBytes());
+        connection = (HttpURLConnection) url.openConnection();
+        connection.setDoOutput(true);
+        connection.setRequestMethod("PUT");
+        connection.setRequestProperty("authorization", "Basic " + base64);
+        connection.setRequestProperty("accept", "application/json");
+        connection.setRequestProperty(
+                "Content-Type", "multipart/form-data; boundary="+BOUNDARY);
+        outStream = new DataOutputStream(this.connection.getOutputStream());
+        writeMultipartBody();
+        outStream.flush();
+        outStream.close();
+        // checks server's status code first
+        int status = this.connection.getResponseCode();
+        if (status == HttpURLConnection.HTTP_CREATED) {
+            InputStream responseStream = new
+                BufferedInputStream(connection.getInputStream());
+            BufferedReader responseStreamReader =
+                new BufferedReader(new InputStreamReader(responseStream));
+            String line = "";
+            StringBuilder stringBuilder = new StringBuilder();
+            while ((line = responseStreamReader.readLine()) != null) {
+                stringBuilder.append(line).append("\n");
+            }
+            responseStreamReader.close();
+            response.setResponseCode(status);
+            response.setBody(stringBuilder.toString());
+            response.setHeaders(this.connection.getHeaderFields());
+            this.connection.disconnect();
+        } else {
+            throw new HyperwalletException("Server returned non-OK status: " + status);
+        }
+        return response;
+    }
+
+    private void buildHeaders() {
+        if (!headers.isEmpty()) {
+            for (Map.Entry<String, List<String>> entry : headers.entrySet()) {
+
+                for (String value : entry.getValue()) {
+                    connection.addRequestProperty(entry.getKey(), value);
+                }
+            }
+        }
+    }
+
+    private void writeMultipartBody() throws IOException {
+        for(Multipart.MultipartData multipartData : multipartList.getMultipartList()) {
+            for (Map.Entry<String, String> entry : multipartData.getEntity().entrySet()) {
+
+                outStream.writeBytes(this.SEPARATOR + this.BOUNDARY + this.CRLF);
+                outStream.writeBytes(multipartData.getContentDisposition());
+                outStream.writeBytes(multipartData.getContentType());
+                outStream.writeBytes(this.CRLF);
+
+                if(multipartData.getContentType().contains("image")){
+                    byte[] bytes = Files.readAllBytes(new File(entry.getValue().toString()).toPath());
+                    outStream.write(bytes);
+                    outStream.writeBytes(this.CRLF);
+                }else{
```

Comment:
weird formatting. Missing spaces in this file

Reply:
corrected the formatting